runtime/caml/domain_state.h

```diff
@@ -44,9 +44,10 @@ enum {
 
 /* Check that the structure was laid out without padding,
    since the runtime assumes this in computing offsets */
-CAML_STATIC_ASSERT(
+static_assert(
```

Comment:
Doesn't this defeat the changes in the other commit? Can you be sure this file will always be compiled in an environment where `static_assert` is available instead of, say, `_Static_assert`? What about MSVC?

Reply:
> Doesn't this defeat the changes in the other commit?

It does if there are environments where `static_assert` isn't available. If they exist, I'll remove this commit. Otherwise I can squash them too.

> Can you be sure this file will always be compiled in an environment where `static_assert` is available instead of, say, `_Static_assert`?

In C11, `static_assert` requires the `assert.h` header. In C23, it's always defined as a keyword. It's tempting to use `_Static_assert` directly, but it's been removed from C23…

> What about MSVC?

You're right, that might be trickier. It requires the `/std:c11` flag, which I'm not sure we're passing, and the Windows SDK 10.0.20348.0 (version 2104) or later (presumably just for the `assert.h` header). On the other hand, MSVC support for trunk has not been reinstated yet. We could define that flag too.

Reply:
No worries: OCaml 5.0 requires at least C11, if only for the low-level atomics, so if MSVC support comes back, it will be in C11 mode.  Feel free to use C11 features!  C23 features gives me pause, as C23 is still a draft and we don't have any pre-C23 testing in the CI at the moment.  (See discussion at #12468.)